### DISCUSSION TITLE
Can i use API about create_folder?

### DISCUSSION THREAD

**Discussion by stormkuter** [rejetto/hfs#396] [Q&A] [answered]
i want to create a directory on my server using api, is it possible?
If possible, how can i use api??

**Comment by stormkuter** [comment]
is not there who know it?

**Comment by rejetto** [comment, marked as answer]
For some reason I didn't get any notification for your message, I'm sorry.
Did you already have a look at the call the web interface is using? Because that's the same you should be using too.
You should first clarify if the folder is to be created on disk, or should be virtual.

I just wrote this page to explain how the VFS works https://github.com/rejetto/hfs/wiki/Virtual-file-system
while I will document APIs on https://github.com/rejetto/hfs/wiki/APIs

**Reply by stormkuter** [reply]
Thanks in advance

I want to create on disk of server.
i tried create_folder to http using Python's request module, but it failed.

example)
url = "http://120.22.22.196:8181/~/api/create_folder"
data = {"uri": "/mnt/dev/", "name": "1313"}
response = requests.post(api, json=data)

return status is 418.

Can you tell me what the problem is?

**Comment by rejetto** [comment]
ah, yes, thanks for reporting. I forgot to document that any POST request must have the following header: `x-hfs-anti-csrf: 1`

**Reply by stormkuter** [reply]
oh thanks!
work well :)

**Comment by stormkuter** [comment]
POST request must have the following header: x-hfs-anti-csrf: 1
